libdevcore/CMakeLists.txt

```diff
@@ -2,7 +2,9 @@ file(GLOB sources "*.cpp")
 file(GLOB headers "*.h")
 
 add_library(devcore ${sources} ${headers})
-target_link_libraries(devcore PRIVATE jsoncpp ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(devcore PRIVATE ${JSONCPP_LIBRARY} ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
@chfast are you sure `jsoncpp` here will be correct and we don't need to use `${JSONCPP_LIBRARY}` here?

Reply:
It's better to use a CMake target representing a library (defined in https://github.com/ethereum/solidity/blob/develop/cmake/jsoncpp.cmake#L47) than the path to the library directly (old school CMake).

The `jsoncpp` defines also include paths, however we have to workaround it with additional `include_directories()` because we want to make sure jsoncpp's include path is before `/usr/include`.

So in other words. both work but using `jsoncpp` is better style.